internal/token/jwx/jwx.go

```diff
@@ -248,3 +253,35 @@ func tokenFromClaims(claims token.Claims) (jwt.Token, error) {
 
 	return t, nil
 }
+
+// GenerateAdminToken generates a token for an admin. The returned token is
+// in JSON format.
+func GenerateAdminToken(ctx context.Context, req *pb.GenerateAdminTokenRequest) (*pb.GenerateAdminTokenResponse, error) {
+	tm := NewTokenManager(HS256)
+
+	// Get the expiration values from config.
+	if req.RefreshExpiration <= 0 {
+		req.RefreshExpiration = int64(24 * time.Hour)
+	}
+	if req.AccessExpiration <= 0 {
+		req.AccessExpiration = int64(30 * time.Minute)
+	}
+
+	// Generate the token.
+	s, err := token.CreateAdminSecret(tm, token.Config{
+		AdminName:         req.AdminName,
+		Subject:           "admin",
+		Roles:             nil,
+		JWTSigningSecret:  JWTSigningSecret,
```

Comment:
How are you getting the user input for the JWT secret for karavictl admin token command?

Reply:
Either with --password, example: `karavictl admin token --name firstdamin --jwt-signing-secret secret`

OR

omit `--jwt-signing-secret`, cli will prompt for password style input prompt.

Reply:
@alikdell yes I see that in the karavictl command, how is that value being passed here? I see `JWTSigningSecret` is set to default value "secret" in this file.

Reply:
fixed. Signing secret is now being passed in